src/sql/workbench/contrib/notebook/browser/notebookExplorer/notebookSearchResultsView.ts

```diff
@@ -0,0 +1,242 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as DOM from 'vs/base/browser/dom';
+import { ActionBar } from 'vs/base/browser/ui/actionbar/actionbar';
+import { CountBadge } from 'vs/base/browser/ui/countBadge/countBadge';
+import { ITreeNode, ITreeRenderer } from 'vs/base/browser/ui/tree/tree';
+import { IAction } from 'vs/base/common/actions';
+import { Disposable, IDisposable, dispose } from 'vs/base/common/lifecycle';
+import * as resources from 'vs/base/common/resources';
+import * as nls from 'vs/nls';
+import { IConfigurationService } from 'vs/platform/configuration/common/configuration';
+import { FileKind } from 'vs/platform/files/common/files';
+import { IInstantiationService } from 'vs/platform/instantiation/common/instantiation';
+import { ISearchConfigurationProperties } from 'vs/workbench/services/search/common/search';
+import { attachBadgeStyler } from 'vs/platform/theme/common/styler';
+import { IThemeService } from 'vs/platform/theme/common/themeService';
+import { IWorkspaceContextService } from 'vs/platform/workspace/common/workspace';
+import { ResourceLabels } from 'vs/workbench/browser/labels';
+import { RemoveAction, ReplaceAllInFolderAction } from 'vs/workbench/contrib/search/browser/searchActions';
+import { FileMatch, Match, RenderableMatch, SearchModel, FolderMatch } from 'vs/workbench/contrib/search/common/searchModel';
+import { NotebookSearchView } from 'sql/workbench/contrib/notebook/browser/notebookExplorer/notebookSearchView';
+import { IFolderMatchTemplate, IFileMatchTemplate, IMatchTemplate } from 'vs/workbench/contrib/search/browser/searchResultsView';
+
+export class FolderMatchRenderer extends Disposable implements ITreeRenderer<FolderMatch, any, IFolderMatchTemplate> {
+	static readonly TEMPLATE_ID = 'folderMatch';
+
+	readonly templateId = FolderMatchRenderer.TEMPLATE_ID;
+
+	constructor(
+		private searchModel: SearchModel,
+		private searchView: NotebookSearchView,
```

Comment:
is the only difference that the searchView for FolderMatchRenderer is a NotebookSearchView instead of SearchView?

Reply:
yes that's pretty much the difference.

Reply:
hmm i'm trying to figure out if NotebookSearchView can extend SearchView (and if we can just overwrite things methods/properties). Potential for less duplication in NotebookSearchView, and we'd be able to not have to redefine this class. Curious what your thoughts are there if that seems doable.